### ISSUE TITLE
Default dtype

### ISSUE THREAD

**Issue by dseuss** [dseuss/mpnum#28]
Is there any reason not to have a real dtype as default value for all the generators? I always get confused since this would be numpy's standard behavior.

**Comment by milan-hl** [comment]
```
In [66]: mp.factory.zero(2, 1, 1).dtype
Out[66]: numpy.float64

In [67]: mp.factory.random_mpa(2, 1, 1).dtype
Out[67]: numpy.complex128

In [68]: mp.factory.random_mps(2, 1, 1).dtype
Out[68]: numpy.complex128

In [69]: mp.factory.random_mpo(2, 1, 1).dtype
Out[69]: numpy.complex128
```

I agree that random_mpa() could be changed to default to real. I think random_mps() returning a vector with real entries would be a bit surprising, but I'm not so sure about random_mpo(). 

However, if we change random_mpa(), I think there are potentially a lot of tests that should be changed because they should also be tested for complex values (compression, arithmenics, MP-POVM related things, ...).

**Comment by dseuss** [comment]
Most of tests test both double precision dtypes anyway. ⁣​So that shouldn't be a problem. 

I agree, let's leave the physics specific mps/mpo functions the way they are and change random_mpa

**Comment by dseuss** [comment]
Fixed in cb625fe
